proposed/http-middleware/middleware.md

```diff
@@ -1,114 +1,123 @@
 
-- `Psr\Http\ServerMiddleware\DelegateInterface`
+- `Psr\Http\Server\MiddlewareInterface`
 
-Legacy middleware implementing a double pass approach MUST be wrapped using an
-object that implements the `MiddlewareInterface`.
+### 1.2 Request Handlers
 
-### 1.1 Dispatchers
+An HTTP request handler is an individual component that processes a request and
+produces a response, as defined by PSR-7.
 
-An HTTP middleware dispatcher is an object that holds multiple middleware
-components that can be used to process one or more requests in sequence.
+Request handlers using this standard MUST implement the following interface:
 
-The middleware dispatcher MUST pass the request and a delegate to each
-middleware for further processing. The delegate MUST be able to dispatch
-the next available middleware or if no more middleware is available, create a
-default response.
+- `Psr\Http\Server\RequestHandlerInterface`
 
-### 1.2 Generating Responses
+### 1.3 Generating Responses
 
-It is RECOMMENDED that any middleware that needs to generate a response will
-use HTTP Factories as defined in [PSR-17](https://github.com/php-fig/fig-standards/tree/master/proposed/http-factory),
-in order to prevent dependence on a specific HTTP message implementation.
+It is RECOMMENDED that any middleware or request handler that generates a response
+will use HTTP factories as defined in [PSR-17][psr17] in order to prevent dependence
+on a specific HTTP message implementation.
+
+[psr17]: https://github.com/php-fig/fig-standards/tree/master/proposed/http-factory
 
 ## 2. Interfaces
 
-### 2.1 Psr\Http\ServerMiddleware\MiddlewareInterface
+### 2.1 Psr\Http\Server\MiddlewareInterface
 
-The following interface MUST be implemented by compatible server middleware components.
+The following interface MUST be implemented by compatible middleware components.
 
 ```php
-namespace Psr\Http\ServerMiddleware;
+namespace Psr\Http\Server;
 
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 
+/**
+ * An HTTP middleware component participates in processing an HTTP message,
+ * either by acting on the request or the response. This interface defines the
+ * methods required to use the middleware.
+ */
 interface MiddlewareInterface
 {
     /**
      * Process an incoming server request and return a response, optionally delegating
-     * to the next middleware component to create the response.
+     * response creation to a handler.
      *
      * @param ServerRequestInterface $request
-     * @param DelegateInterface $delegate
+     * @param RequestHandlerInterface $handler
      *
      * @return ResponseInterface
      */
     public function process(
         ServerRequestInterface $request,
-        DelegateInterface $delegate
+        RequestHandlerInterface $handler
     );
 }
 ```
 
-### 2.2 Psr\Http\ServerMiddleware\DelegateInterface
+### 2.2 Psr\Http\Server\RequestHandlerInterface
```

Comment:
Same comment here as for sections 1.1/1.2 - invert these, so that the `RequestHandlerInterface` is introduced first.